src/atoms/RadioButton/RadioButton.native.js

```diff
@@ -0,0 +1,363 @@
+      }
+    },
+    size(checkboxSize) {
+      switch (checkboxSize) {
+        case 'large':
+          return 'medium';
+        case 'medium':
+          return 'xsmall';
+        default:
+          return 'small';
+      }
+    },
+  },
+  title: {
+    margin() {
+      return [0, 0, 0, 0.5];
+    },
+  },
+  backdrop: {
+    dimensions(size) {
+      switch (size) {
+        case 'large':
+          return {
+            width: '28px',
+            height: '28px',
+            borderRadius: '14px',
+          };
+        case 'medium':
+          return {
+            width: '24px',
+            height: '24px',
+            borderRadius: '12px',
+          };
+        case 'small':
+          return {
+            width: '20px',
+            height: '20px',
+            borderRadius: '10px',
+          };
+
+        default:
+          return {
+            width: '24px',
+            height: '24px',
+            borderRadius: '12px',
+          };
+      }
+    },
+  },
+  circle: {
+    dimensions(size) {
+      switch (size) {
+        case 'large':
+          return {
+            width: '14px',
+            height: '14px',
+            borderRadius: '7px',
+            borderWidth: '1.4px',
+          };
+        case 'medium':
+          return {
+            width: '12px',
+            height: '12px',
+            borderRadius: '6px',
+            borderWidth: '1.2px',
+          };
+        case 'small':
+          return {
+            width: '10px',
+            height: '10px',
+            borderRadius: '5px',
+            borderWidth: '1px',
+          };
+        default:
+          return {
+            width: '12px',
+            height: '12px',
+            borderRadius: '6px',
+            borderWidth: '1.2px',
+          };
+      }
+    },
+  },
+  dot: {
+    dimensions(size) {
+      switch (size) {
+        case 'large':
+          return {
+            width: '7px',
+            height: '7px',
+            borderRadius: '3.5px',
+          };
+        case 'medium':
+          return {
+            width: '6px',
+            height: '6px',
+            borderRadius: '3px',
+          };
+        case 'small':
+          return {
+            width: '5px',
+            height: '5px',
+            borderRadius: '2.5px',
+          };
+        default:
+          return {
+            width: '6px',
+            height: '6px',
+            borderRadius: '3px',
+          };
+      }
+    },
+  },
+};
+
+const Dot = styled(View)(
+  (props) =>
+    `
+    border-radius: ${props.borderRadius};
+    background-color: ${props.backgroundColor};
+  `,
+);
+
+const Circle = styled(View)(
+  (props) =>
+    `
+    border-radius: ${props.borderRadius}; 
+    border-color: ${props.color};
+    border-width: ${props.borderWidth}; 
+  `,
+);
+
+const isChecked = (context, currentValue) => {
+  return context && isDefined(context.value) && context.value === currentValue;
+};
+
+const RadioButtonContext = React.createContext(null);
```

Comment:
Refer [this](https://github.com/kamleshchandnani/design-system/blob/master/src/molecules/Modal/ModalContext.js) for Context creation